Mathlib/Topology/Algebra/Module/WeakDual.lean

```diff
@@ -326,6 +353,64 @@ theorem coe_map (f : E →L[𝕜] F) : (WeakSpace.map f : E → F) = f :=
   rfl
 #align weak_space.coe_map WeakSpace.coe_map
 
+/-- The canonical preimage in `E` of an open set in `WeakSpace 𝕜 E` is open in `E`. -/
+theorem isOpen_of_isOpen_WeakSpace (U : Set (WeakSpace 𝕜 E))
+    (hU : IsOpen[(WeakSpace.instTopologicalSpace : TopologicalSpace (WeakSpace 𝕜 E))] U) :
+    IsOpen ((continuousLinearMapToWeakSpace 𝕜 E) ⁻¹' U) := by
+  exact ((continuousLinearMapToWeakSpace 𝕜 E).cont).isOpen_preimage U hU
+
+/-- A set in `E` which is open in the weak topology is open. -/
+theorem isOpen_of_isOpen_WeakSpace' (V : Set E)
+    (hV : IsOpen[(WeakSpace.instTopologicalSpace)]
+    ((continuousLinearMapToWeakSpace 𝕜 E) '' V : Set (WeakSpace 𝕜 E))) : IsOpen V := by
+  have : (continuousLinearMapToWeakSpace 𝕜 E) ⁻¹' ((continuousLinearMapToWeakSpace 𝕜 E) '' V)
+  = V := by
+   ext x
+   constructor
+   · intro hx
+     obtain ⟨y, hy⟩ := hx
+     rw [toWeakSpace_eq_continuousLinearMapToWeakSpace,
+       toWeakSpace_eq_continuousLinearMapToWeakSpace] at hy
+     have : y = x := LinearEquiv.injective (toWeakSpace 𝕜 E) hy.2
+     rw [this] at hy
+     exact hy.1
+   · intro hx
+     simp only [Set.mem_preimage, Set.mem_image]
+     use x
+  rw [← this]
+  exact isOpen_of_isOpen_WeakSpace ((continuousLinearMapToWeakSpace 𝕜 E) '' V) hV
```

Comment:
```suggestion
  rw [← Equiv.image_preimage (toWeakSpace 𝕜 E).toEquiv V]
  exact isOpen_of_isOpen_WeakSpace ((toWeakSpace 𝕜 E) '' V) hV
```

Reply:
This code gives an error. Indeed, it seems to try to prove that `(continuousLinearMapToWeakSpace 𝕜 E)` is an open map, which is not. See above.

Reply:
As you didn't tell me to remove this, I assume that this is needed.